source/core/module_coreHelpers.js

```diff
@@ -56,15 +56,18 @@
             fpSync = fn;
         };
 
-        Module.coreHelpers.utf8ArrayToStringWithNull = function (u8Array, idxIn) {
+        // Takes Vireo Strings that are UTF-8 encoded strings with known length and returns a JS string
+        Module.coreHelpers.sizedUtf8ArrayToJSString = function (u8Array, startIndex, length) {
             /* eslint-disable no-continue, no-plusplus, no-bitwise */
             var u0, u1, u2, u3, u4, u5;
-            var idx = idxIn;
+            var idx = startIndex;
+            var lastVisitedIndex = startIndex + length;
```

Comment:
Never mind.  It's clear enough, I think my confusion came from thinking in terms of C APIs, but in this case the UTF8Array knows its length, and doesn't need to be passed.

However, there are still some problematic corner cases:
The code will read one more byte than it should of u8Array. For example, if the array is empty, or you pass length of 0, it will still read the first element before returning.  It doesn't assign it to str, but it still reads past the end of the array.   Just make the test >= and move it before the array read.
Also, if you pass a length in a middle of a UTF8 code point, it will read more bytes than specified, including past the end of the array if the array isn't well-formed Unicode.